File: day1/1.c
#include <stddef.h>
#include <stdio.h>

int main() {
    FILE* f = fopen("./input.txt","r");
    int buf[2048];
    size_t read = 0;
    int curr = 0;
    while ((read = fread(buf, sizeof(int), 2048, f)) > 0) {
        for (size_t i = 0; i < read; i++) {
            printf("buf[i]: %d\n", buf[i]);
        } 
    }
}

File: day1/1.py
f = open("./input.txt", "r")
curr = 0
top = [0]*3
for num in f.readlines():
    if num == '\n':
        if curr > top[-1]:        
            top.append(curr)
        elif curr > top[-2]:
            top.insert(-2, curr)
        elif curr > top[-3]:
            top.insert(-3, curr)
        curr = 0
    else:
        curr += int(num)

ret = top.pop() + top.pop() + top.pop()
print(ret)
